modules/core/shared/src/main/scala/lucuma/core/util/TimeSpan.scala

```diff
@@ -141,6 +144,39 @@ object TimeSpan {
     def format: String =
       toDuration.toString
 
+    /**
+     * Adds two TimeSpan values, capping the resulting value at `Max`.
+     */
+    @targetName("boundedAdd")
+    def +|(other: TimeSpan): TimeSpan =
+      fromMicroseconds(timeSpan.toMicroseconds + other.toMicroseconds).getOrElse(Max)
+
+    /**
+     * Subtracts a TimeSpan value, with a floor of `Min` on the resulting value.
+     */
+    @targetName("boundedSubtract")
+    def -|(other: TimeSpan): TimeSpan =
+      fromMicroseconds(timeSpan.toMicroseconds - other.toMicroseconds).getOrElse(Min)
+
+    /**
+     * Multiplies a TimeSpan by an integer, limiting the resulting value to the
+     * range (`Min`, `Max`).
+     */
+    @targetName("boundedMultiply")
+    def *|(multiplier: Int): TimeSpan = {
```

Comment:
Is there a reason for division to require a `PosInt` but multiplication to allow negative numbers?

Also, can we support decimals (with rounding on the result) ?

Reply:
Actually I just want a `NotZeroInt` but I don't think that exists?  Dividing by a negative is never going to produce anything but `Zero` so I went with `PosInt`.

Decimal options with rounding seem fine.  I just didn't need them for what I'm working on.

Reply:
Won't multiplying by a negative also always produce `Zero` ?

Reply:
I imagine with `refined` you can scare up a `Not[Equal[_0]]` or some such but it seems awkward, I don't know.

Reply:
> Won't multiplying by a negative also always produce Zero ?

This is true.  It could be a `NonNegInt` but it's easier to work with a raw `Int` and there's a valid way to use the full range of `Int` as the multiplier so I switched to that.  I guess the inconsistency is a little irksome though.